src/emulator/storage/files.ts

```diff
@@ -162,6 +182,37 @@ export class StorageLayer {
     return [...this._buckets.values()];
   }
 
+  /**
+   * Returns an stored object and its metadata.
+   * @throws {NotFoundError} if object does not exist
+   * @throws {ForbiddenError} if request is unauthorized
+   */
+  public async handleGetObject(request: GetObjectRequest): Promise<GetObjectResponse> {
+    const operationPath = ["b", request.bucketId, "o", request.decodedObjectId].join("/");
+    const metadata = this.getMetadata(request.bucketId, request.decodedObjectId);
+
+    let authorized = (metadata?.downloadTokens || []).includes(request.downloadToken ?? "");
+    if (!authorized) {
+      authorized = await isPermitted({
+        ruleset: this._rules,
+        method: RulesetOperationMethod.GET,
+        path: operationPath,
+        file: { before: metadata?.asRulesResource() },
+        authorization: request.authorization,
+      });
+    }
+    if (!authorized) {
```

Comment:
I think it's worth leaving in the comments about auth via header vs. token for web vs. mobile